hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/common/HoodieSparkEngineContext.java

```diff
@@ -214,4 +214,17 @@ public List<Integer> removeCachedDataIds(HoodieDataCacheKey cacheKey) {
       return removed == null ? Collections.emptyList() : removed;
     }
   }
+
+  @Override
+  public Map<String, String> getInfo() {
+    Map<String, String> info = new HashMap<>();
+    info.put("spark.user", javaSparkContext.sparkUser());
+    info.put("spark.master", javaSparkContext.master());
+    info.put("spark.application", javaSparkContext.appName());
+    info.put("spark.version", javaSparkContext.version());
+    info.put("spark.defaultParallelism", String.valueOf(javaSparkContext.defaultParallelism()));
+    info.put("spark.defaultMinPartitions", String.valueOf(javaSparkContext.defaultMinPartitions()));
+    info.put("spark.executor.instances", String.valueOf(javaSparkContext.getConf().get("spark.executor.instances")));
+    return info;
```

Comment:
should we also let users configure any hoodie write configs to be added to the extra metadata. since we don't serialize them anywhere, sometimes might come in handy during investigations. by default we don't need to add any hoodie write configs. but if user configures them, we can add them as well

Reply:
Simply easier to add the entire hoodie config (without schema).